demo-app/src/commonMain/kotlin/dev/sargunv/maplibrecompose/demoapp/demos/SnapshotterDemo.kt

```diff
@@ -0,0 +1,151 @@
+import dev.sargunv.maplibrecompose.demoapp.DEFAULT_STYLE
+import dev.sargunv.maplibrecompose.demoapp.Demo
+import dev.sargunv.maplibrecompose.demoapp.DemoMapControls
+import dev.sargunv.maplibrecompose.demoapp.DemoOrnamentSettings
+import dev.sargunv.maplibrecompose.demoapp.DemoScaffold
+import kotlin.coroutines.cancellation.CancellationException
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+
+object SnapshotterDemo : Demo {
+  override val name = "Snapshotter"
+  override val description = "Take a snapshot of the map"
+
+  @Composable
+  override fun Component(navigateUp: () -> Unit) {
+    val cameraState = rememberCameraState()
+    val styleState = rememberStyleState()
+    val isLoading = remember { mutableStateOf(false) }
+    val snapshot = remember { mutableStateOf<ImageBitmap?>(null) }
+    val lifeCycleOwner = LocalLifecycleOwner.current
+
+    DisposableEffect(lifeCycleOwner) {
+      val observer = LifecycleEventObserver { _, event ->
+        if (event == Lifecycle.Event.ON_PAUSE) {
+          isLoading.value = false
+        }
+      }
+      lifeCycleOwner.lifecycle.addObserver(observer)
+      onDispose { lifeCycleOwner.lifecycle.removeObserver(observer) }
+    }
+
+    DemoScaffold(this, navigateUp) {
+      Column {
+        Box(modifier = Modifier.weight(1f)) {
+          MaplibreMap(
+            styleUri = DEFAULT_STYLE,
+            cameraState = cameraState,
+            styleState = styleState,
+            ornamentSettings = DemoOrnamentSettings(),
+          )
+          DemoMapControls(cameraState, styleState)
+
+          if (isLoading.value) {
+            CircularProgressIndicator(modifier = Modifier.align(Alignment.Center))
+          }
+        }
+
+        SnapshotterControls(cameraState, isLoading, snapshot)
+
+        snapshot.value?.let {
+          SnapshotDialog(snapshot = it, onDismissRequest = { snapshot.value = null })
+        }
+      }
+    }
+  }
+
+  @Composable
+  private fun SnapshotterControls(
+    cameraState: CameraState,
+    isLoading: MutableState<Boolean>,
+    snapshot: MutableState<ImageBitmap?>,
+  ) {
+    val scope = rememberCoroutineScope()
+    var snapshotJob by remember { mutableStateOf<Job?>(null) }
+
+    Row(
+      modifier = Modifier.padding(16.dp).fillMaxWidth(),
+      horizontalArrangement = Arrangement.SpaceEvenly,
+    ) {
+      Button(
+        onClick = {
+          snapshotJob =
+            scope.launch {
+              isLoading.value = true
+              try {
+                val response =
+                  cameraState.snapshot(
+                    width = 500.dp,
+                    height = 500.dp,
+                    styleUri = DEFAULT_STYLE,
+                    cameraPosition = cameraState.position,
+                  )
+
+                snapshot.value = response
+              } catch (error: SnapshotException) {
+                println("Error during snapshot generation: ${error.message}")
```

Comment:
ideally the demo should show a dialog (I just got a timeout b/c I tried the snapshot on slow train internet, and it was unclear why it stopped loading)